locallib.php

```diff
@@ -459,6 +535,9 @@ private function process_action_delete_choice() {
                 if ($choice) {
                     $DB->delete_records(this_db\ratingallocate_group_choices::TABLE, ['choiceid' => $choiceid]);
                     $DB->delete_records(this_db\ratingallocate_choices::TABLE, array('id' => $choiceid));
+                    // Delete related group associations, if any.
+                    $DB->delete_records(this_db\ratingallocate_group_choices::TABLE, array('choiceid' => $choiceid));
```

Comment:
Yes, thanks for spotting. I guess the issue with missing deletion you found in the groups PR has been fixed here, so this is duplicated, because I fixed it separately in the groups PR and did not rebase carefully enough.